app/validators/url_format_validator.rb

```diff
@@ -8,7 +8,13 @@ def validate_each(record,attribute,value)
     # http (optional s) :// domain . tld (optional port) / anything
     regexp = /^https?:\/\/[_a-z\d\-]+\.[._a-z\d\-]+(:\d+)?\/?.+/i
     unless value.match regexp
-      record.errors.add(attribute, options[:message] || "does not appear to be a valid URL.")
+      record.errors.add(attribute, options[:message] || :invalid)
+    end
+
+    begin
+      Addressable::URI.heuristic_parse(value)
+    rescue Addressable::URI::InvalidURIError
+      record.errors.add(attribute, options[:message] || :invalid)
```

Comment:
I don't like this control flow with the repeated errors.add but could not come up with anything better since it's using a `rescue` :/